library/Icingadb/Model/RedundancyGroupParentStateSummary.php

```diff
@@ -0,0 +1,110 @@
+                'SUM(CASE WHEN redundancy_group_from_to_host_state.is_problem = \'y\''
+                . ' AND redundancy_group_from_to_host_state.is_acknowledged = \'n\' THEN 1 ELSE 0 END'
+                . '+ CASE WHEN redundancy_group_from_to_service_state.is_problem = \'y\''
+                . ' AND redundancy_group_from_to_service_state.is_acknowledged = \'n\' THEN 1 ELSE 0 END)'
+            ),
+            'objects_total' => new Expression(
+                'SUM(CASE WHEN redundancy_group_from_to_host.id IS NOT NULL THEN 1 ELSE 0 END)'
+                . '+ SUM(CASE WHEN redundancy_group_from_to_service.id IS NOT NULL THEN 1 ELSE 0 END)'
+            ),
+            'objects_ok' => new Expression(
+                'SUM(CASE WHEN redundancy_group_from_to_host_state.soft_state = 0 THEN 1 ELSE 0 END'
+                . '+ CASE WHEN redundancy_group_from_to_service_state.soft_state = 0 THEN 1 ELSE 0 END)'
+            ),
+            'objects_unknown_handled' => new Expression(
+                'SUM(CASE WHEN redundancy_group_from_to_service_state.soft_state = 3'
+                . ' AND (redundancy_group_from_to_service_state.is_handled = \'y\''
+                . ' OR redundancy_group_from_to_service_state.is_reachable = \'n\') THEN 1 ELSE 0 END)'
+            ),
+            'objects_unknown_unhandled' => new Expression(
+                'SUM(CASE WHEN redundancy_group_from_to_service_state.soft_state = 3'
+                . ' AND redundancy_group_from_to_service_state.is_handled = \'n\''
+                . ' AND redundancy_group_from_to_service_state.is_reachable = \'y\' THEN 1 ELSE 0 END)'
+            ),
+            'objects_warning_handled' => new Expression(
+                'SUM(CASE WHEN redundancy_group_from_to_service_state.soft_state = 1'
+                . ' AND (redundancy_group_from_to_service_state.is_handled = \'y\''
+                . ' OR redundancy_group_from_to_service_state.is_reachable = \'n\') THEN 1 ELSE 0 END)'
+            ),
+            'objects_warning_unhandled' => new Expression(
+                'SUM(CASE WHEN redundancy_group_from_to_service_state.soft_state = 1'
+                . ' AND redundancy_group_from_to_service_state.is_handled = \'n\''
+                . ' AND redundancy_group_from_to_service_state.is_reachable = \'y\' THEN 1 ELSE 0 END)'
+            )
+        ];
+    }
+
+    public static function on(Connection $db)
+    {
+        $q = parent::on($db)->with([
+            'from',
+            'from.to.host',
+            'from.to.host.state',
+            'from.to.service',
+            'from.to.service.state'
+        ]);
+
+        /** @var static $m */
+        $m = $q->getModel();
+        $q->columns($m->getSummaryColumns());
+
+        return $q;
+    }
+
+    public function getColumns(): array
+    {
+        return array_merge(parent::getColumns(), $this->getSummaryColumns());
+    }
+
+    public function getDefaultSort()
```

Comment:
Please add return type.

Reply:
IMO, I think there is no need to add return type to this method. But, now that I look at it, I don't think it is necessary to have this method at all.